### PR TITLE
Add random access feature to existing netCDF parser

### PR DESCRIPTION
**Pull request by adhi1419**

### CODE REVIEW

File: msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFFileImportMethod.java [hunk 1]
```diff
@@ -82,7 +85,12 @@
    * @param dataStore a {@link io.github.msdk.datamodel.datastore.DataPointStore} object.
    */
   public NetCDFFileImportMethod(@Nonnull File sourceFile) {
+    this(sourceFile, null);
+  }
+
+  public NetCDFFileImportMethod(@Nonnull File sourceFile, Predicate<MsScan> msScanPredicate) {
     this.sourceFile = sourceFile;
+    this.msScanPredicate = msScanPredicate != null ? msScanPredicate : s -> true;
```
Comment (TDI-tenderholt) [inline msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFFileImportMethod.java]: Can you avoid explicitly passing in a ```null``` value? What about initializing ```msScanPredicate``` to ```s -> true``` (line 68) and then using the ```Predicate.and()``` method?

File: msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    this.intensityValueVariable = intensityValueVariable;
+    this.massValueScaleFactor = massValueScaleFactor;
+    this.intensityValueScaleFactor = intensityValueScaleFactor;
+    this.mzValues = null;
+    this.intensityValues = null;
+    this.spectrumType = null;
+  }
+
+  @Override
+  public float[] getIntensityValues(float[] intensityValues) {
```
Comment (tomas-pluskal) [inline msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java]: If the Predicate was set for the scan, this function should return the pre-loaded data

File: msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java [hunk 3]
```diff
@@ -54,51 +54,70 @@ public NetCDFMsScan(Integer scanNumber, int[] scanStartPositions, float[] scanRe
...
+        for (int i = 0; i < numOfDataPoints; i++) {
+          final Index intensityIndex0 = intensityValuesIndex.set0(i);
+          intensityValues[i] =
+              (float) (intensityValueArray.getDouble(intensityIndex0) * intensityValueScaleFactor);
+        }
+      } catch (IOException | InvalidRangeException e) {
+        throw new MSDKRuntimeException(e);
+      }
+    } else {
       if (intensityValues == null || intensityValues.length < numOfDataPoints)
```
Comment (tomas-pluskal) [inline msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java]: is the intensityValues == null condition here necessary?

File: msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java [hunk 4]
```diff
@@ -54,51 +54,70 @@ public NetCDFMsScan(Integer scanNumber, int[] scanStartPositions, float[] scanRe
...
   public float[] getIntensityValues(float[] intensityValues) {
-    final Integer scanIndex = getScanIndex();
-    numOfDataPoints = getNumberOfDataPoints();
-    try {
-      final int scanStartPosition[] = {scanStartPositions[scanIndex]};
-      final int scanLength[] = {scanStartPositions[scanIndex + 1] - scanStartPositions[scanIndex]};
-      final Array intensityValueArray = intensityValueVariable.read(scanStartPosition, scanLength);
-      final Index intensityValuesIndex = intensityValueArray.getIndex();
-
+    if (this.intensityValues == null) {
```
Comment (tomas-pluskal) [inline msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java]: maybe better to rename this.intensityValues to something like this.preloadedIntensityValues, for clarity

File: msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java (hunk 3 above)
Comment (adhi1419) [inline msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java]: The `getIntensityValues()` function calls this function as `getIntensityValues(null)`. So, it is necessary to check if the passed array is null before checking if the data fits into it.

**Pull request merged**